architecture/evm/block_ref.go

```diff
@@ -345,9 +356,11 @@ func parseCompositeBlockParam(param interface{}) (string, int64, error) {
 				}
 				blockNumber = bni
 			}
-		} else {
-			// Block tag ('latest', 'earliest', 'pending')
+		} else if isBlockTag(v) {
+			// Named block tag (latest, pending, safe, finalized, earliest)
 			blockRef = v
+		} else {
+			return "", 0, errors.New("unsupported block reference: " + v)
```

Comment:
We better be less strict here as for chains or tags that might be added in future.

Probably the best way to solve this issue is simply changing here: https://github.com/erpc/erpc/blob/1016e7d7b4b8ee0be0db64058aaa3ab9a93dc664/architecture/evm/json_rpc_cache.go#L579 to Realtime vs Finalized

Reply:
Although needs more unit tests and actual tests 👌

Reply:
yeah makes sense. thought this is an improvement but it is a bit strict and less ideal to throw an error here.

Reply:
Let's have these unit test scenarios at least (add others if you find more edge-cases):

* User asks for "finalized" tag it must be cached with "realtime" finality flag
* User asks for hex number of finalized block exactly/immediately after we have polled that finalized block, it must be cached with "finalized" finality flag
* WHEN data is cached under realtime finality flag and is not expired yet THEN user requests with the hex number => we must hit the cache ideally
* WHEN data is cached under 'finalized' flag (via hex number) i.e. not realtime flag, THEN user requests with 'finalized' tag => we must NOT hit the cache